Pinta.Gui.Widgets/Widgets/StatusBarColorPaletteWidget.cs

```diff
@@ -277,21 +298,49 @@ private void Palette_ColorChanged (object? sender, EventArgs e)
 			QueueDraw ();
 	}
 
-	private static Color GetUserChosenColor (Color initialColor, string title)
+	public static ColorPickerDialog? active_color_picker;
+	private static void RunColorPicker ()
+	{
+		if (active_color_picker != null)
+			return;
+		active_color_picker = new ColorPickerDialog (PintaCore.Chrome,
+			new[] { PintaCore.Palette.PrimaryColor, PintaCore.Palette.SecondaryColor },
+			0, true);
+		active_color_picker.Show ();
+		active_color_picker.OnResponse += (sender, args) => {
+			if (args.ResponseId == (int) Gtk.ResponseType.Ok) {
+				if (PintaCore.Palette.PrimaryColor != active_color_picker.colors[0])
+					PintaCore.Palette.PrimaryColor = active_color_picker.colors[0];
+				if (PintaCore.Palette.SecondaryColor != active_color_picker.colors[1])
+					PintaCore.Palette.SecondaryColor = active_color_picker.colors[1];
+			}
+
+			active_color_picker = null;
+		};
+	}
+
+
+	private static Color[]? GetUserChosenColor (Color[] colors, int selectedColorIndex = 0, string title = "Color Picker")
 	{
-		var ccd = Gtk.ColorChooserDialog.New (title, PintaCore.Chrome.MainWindow);
-		ccd.UseAlpha = true;
-		ccd.SetColor (initialColor);
+		ColorPickerDialog dialog = new ColorPickerDialog (PintaCore.Chrome, colors, selectedColorIndex, false, title, true);
 
-		Cairo.Color result = initialColor;
+		dialog.OnResponse += (_, args) => dialog.Destroy ();
 
-		var response = ccd.RunBlocking ();
-		if (response == Gtk.ResponseType.Ok)
-			ccd.GetColor (out result);
+		dialog.Show ();
 
-		ccd.Destroy ();
+		var response = dialog.RunBlocking ();
+		if (response == Gtk.ResponseType.Ok) {
+			Color[]? result = new Color[colors.Length];
```

Comment:
I think it would also be possible to just do
```
Color[] result = dialog.colors;
dialog.Destroy ();
return result;
```
unless it's necessary to make a deep copy here?